File: nom-api/Nom.Orch/Models/Shopping/ShoppingListCreateResponseModel.cs
// File: Nom.Orch/Models/Shopping/ShoppingListCreateResponseModel.cs

namespace Nom.Orch.Models.Shopping
{
    public class ShoppingListCreateResponseModel
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public long AuthorId { get; set; }
        public long? HouseholdId { get; set; }
        public long? GroupId { get; set; }
        public DateTime CreatedDate { get; set; }
    }
} 

File: nom-ui/src/app/shopping/models/shopping-list-create-response.model.ts
// File: nom-ui/src/app/shopping/models/shopping-list-create-response.model.ts

import { IShoppingListCreateResponseModel } from './shopping-list-create-response.model.interface';

export class ShoppingListCreateResponseModel implements IShoppingListCreateResponseModel {
    id = 0;
    householdId = 0;
    name = '';
    description?: string;
    createdDate: Date = new Date();
    itemCount = 0;
    completedItemCount = 0;

    constructor(data?: Partial<IShoppingListCreateResponseModel>) {
        if (data) {
            Object.assign(this, data);
        }
    }
} 